scripts/entities/actors/enemy/actoren.gd

```diff
@@ -0,0 +1,21 @@
+extends Actor
+var path: Array = []
+var _actions: int = actions
+
+func _ready() -> void:
+	super()
+	WorldTurnBase.state.next_state.connect(func(world_state):
+		if world_state == turn_state:
+			path = WorldPathfinder.calculate_path(position, WorldTurnBase.players[0].position)
+			path.pop_front()
+			_actions = actions
+			$Timer2.start()
```

Comment:
this is just a pet peeve but I'm a stickler for never referencing things like this (doesn't stop me from doing it because I'm lazy). Instead maybe make an @.onready for it? I leave it up to you however.

Reply:
ah sorry, 5 minute example code, will make it look nice.